### PR TITLE
flexsubscription

### PR DESCRIPTION
**Pull request by horadla23**

### CODE REVIEW

File: core/src/main/java/com/sequenceiq/cloudbreak/controller/v4/FlexSubscriptionV4Controller.java [hunk 1]
```diff
@@ -38,53 +39,55 @@
     private RestRequestThreadLocalService restRequestThreadLocalService;
 
     @Inject
-    private FlexSubscriptionRequestToFlexSubscriptionConverter toFlexSubscriptionConverter;
+    private FlexSubscriptionV4RequestToFlexSubscriptionConverter toFlexSubscriptionConverter;
 
     @Inject
-    private FlexSubscriptionToJsonConverter toJsonConverter;
+    private FlexSubscriptionToFlexSubscriptionV4ResponseConverter toJsonConverter;
```
Comment (topolyai5) [inline core/src/main/java/com/sequenceiq/cloudbreak/controller/v4/FlexSubscriptionV4Controller.java]: use common converterService or converterUtil

File: core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/flexsubscriptions/FlexSubscriptionV4Endpoint.java [hunk 2]
```diff
@@ -25,51 +24,51 @@
...
-    FlexSubscriptionResponse getByNameInWorkspace(@PathParam("workspaceId") Long workspaceId, @PathParam("name") String name);
+    FlexSubscriptionV4Response get(@PathParam("workspaceId") Long workspaceId, @PathParam("name") String name);
 
     @POST
     @Path("")
     @Produces(MediaType.APPLICATION_JSON)
     @ApiOperation(value = FlexSubOpDescription.CREATE_IN_WORKSPACE, produces = ContentType.JSON, notes = Notes.FLEX_SUBSCRIPTION_NOTES,
             nickname = "createFlexSubscriptionInWorkspace")
-    FlexSubscriptionResponse createInWorkspace(@PathParam("workspaceId") Long workspaceId, @Valid FlexSubscriptionRequest request);
+    FlexSubscriptionV4Response create(@PathParam("workspaceId") Long workspaceId, @Valid FlexSubscriptionV4Request request);
```
Comment (doktoric) [inline core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/flexsubscriptions/FlexSubscriptionV4Endpoint.java]: post

File: core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/flexsubscriptions/FlexSubscriptionV4Endpoint.java [hunk 3]
```diff
@@ -25,51 +24,51 @@
...
     @Path("{name}")
     @Produces(MediaType.APPLICATION_JSON)
     @ApiOperation(value = FlexSubOpDescription.DELETE_BY_NAME_IN_WORKSPACE, produces = ContentType.JSON, notes = Notes.FLEX_SUBSCRIPTION_NOTES,
             nickname = "deleteFlexSubscriptionInWorkspace")
-    FlexSubscriptionResponse deleteInWorkspace(@PathParam("workspaceId") Long workspaceId, @PathParam("name") String name);
+    FlexSubscriptionV4Response delete(@PathParam("workspaceId") Long workspaceId, @PathParam("name") String name);
 
     @PUT
-    @Path("setusedforcontroller/{name}")
+    @Path("{name}/set_used")
```
Comment (doktoric) [inline core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/flexsubscriptions/FlexSubscriptionV4Endpoint.java]: set_for_controller?

File: core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/flexsubscriptions/FlexSubscriptionV4Endpoint.java [hunk 4]
```diff
@@ -25,51 +24,51 @@
...
     @Produces(MediaType.APPLICATION_JSON)
     @ApiOperation(value = FlexSubOpDescription.SET_USED_FOR_CONTROLLER_IN_WORKSPACE, produces = ContentType.JSON, notes = FLEX_SUBSCRIPTION_NOTES,
             nickname = "putUsedForControllerFlexSubscriptionByNameInWorkspace")
-    void setUsedForControllerInWorkspace(@PathParam("workspaceId") Long workspaceId, @PathParam("name") String name);
+    FlexSubscriptionV4Response setUsedForController(@PathParam("workspaceId") Long workspaceId, @PathParam("name") String name);
 
     @PUT
-    @Path("setdefault/{name}")
+    @Path("{name}/set_default")
```
Comment (doktoric) [inline core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/flexsubscriptions/FlexSubscriptionV4Endpoint.java]: default?

**Pull request merged**